data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/tekuv1/admin/Readiness.java

```diff
@@ -60,10 +79,27 @@ public Readiness(final DataProvider provider) {
   @Override
   public void handle(final Context ctx) throws Exception {
     ctx.header(Header.CACHE_CONTROL, CACHE_NONE);
-    if (!chainDataProvider.isStoreAvailable() || syncProvider.isSyncing()) {
+    if (!chainDataProvider.isStoreAvailable()
+        || syncProvider.isSyncing()
+        || !reachedTargetPeerCount(ctx)) {
       ctx.status(SC_SERVICE_UNAVAILABLE);
     } else {
       ctx.status(SC_OK);
     }
   }
+
+  private boolean reachedTargetPeerCount(final Context ctx) {
+    try {
+      final Map<String, List<String>> parameters = ctx.queryParamMap();
+      if (parameters.isEmpty()) return true; // Backwards compatibility
+      LOG.debug("Current peer count is {}", networkDataProvider.getPeerCount());
+      return !parameters.isEmpty()
+          && parameters.containsKey(TARGET_PEER_COUNT)
+          && networkDataProvider.getPeerCount()
+              > getParameterValueAsLong(parameters, TARGET_PEER_COUNT);
+    } catch (final IllegalArgumentException ex) {
+      LOG.trace("Illegal parameter in Readiness", ex);
```

Comment:
This should probably result in a bad request response being given so the caller has a way to tell that the request couldn't be parsed, rather than just always thinking the node wasn't ready.

Reply:
Good point, I'll update it.